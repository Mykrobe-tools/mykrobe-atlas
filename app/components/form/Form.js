/* @flow */

import React, { Component, PropTypes } from 'react';

class Form extends Component {

  render() {
    const {children, onSubmit} = this.props;
    return (
      <form onSubmit={(event) => onSubmit(event)}>
        {children}
      </form>
    );
  }
}

Form.propTypes = {
  children: PropTypes.node.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Form;
