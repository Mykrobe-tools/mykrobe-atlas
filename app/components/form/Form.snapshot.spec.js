import React from 'react';
import renderer from 'react-test-renderer';
import Form from './Form';

describe('Form component snapshot', () => {
  const children = '<p>Lorem ipsum</p>';
  const onSubmit = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <Form
        children={children}
        onSubmit={onSubmit} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('runs submit event', () => {
    const component = renderer.create(
      <Form
        children={children}
        onSubmit={onSubmit} />
    );
    let tree = component.toJSON();
    tree.props.onSubmit();
    expect(onSubmit).toBeCalled();
  })
});
