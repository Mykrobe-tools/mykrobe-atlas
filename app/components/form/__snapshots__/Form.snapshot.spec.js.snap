exports[`Form component snapshot renders correctly 1`] = `
<form
  onSubmit={[Function]}>
  <p>Lorem ipsum</p>
</form>
`;
